doc/concepts.md

```diff
@@ -1,190 +1,267 @@
+(see below).
+The data models can represent different elements of the software, and
+be used in handling I/O as well as in code generation and
+documentation.
+Data models can also be used for annotating data and data sets.
+This might be useful in cases where for instance the origin of the
+data, license and ownership are of importance.
+
+Since any complex software will have many data models and often multiple
+instances of the same data model, DLite allows for creating collections of
+data models with defined relationships.
+
+One idea of SOFT is that software may be written is such way that
+business logic is handled by the codebase, while I/O, file-formats,
+version handling, data import/export and interoperability can be
+handled by reusable components in the DLite-framework, thus reducing
+risk and development time.
+
+
+Datamodel hierarchy
+-------------------
+
+![Datamodel hierarchy](figs/datamodel-hierarchy.svg)
+
+_**Figure 2.** Datamodel hierarchy.
+The instances colored orange are predefined, while the other may be defined by the user.
+Data instances 5 and 6 are what we normally will call a collection._
+
+Figure 2 shows the datamodel hierarchy implemented in DLite.
+As a user, you will almost always deal with only entities or data
+instances, but the hierarchy gives DLite a strong and theoretically
+well-defined schema.
+
+The actual data or *Data instances* are instances of the *Entity* that
+describes them.
+*Entities* are instances of the *EntitySchema* The *EntitySchema* is
+an instance of the *BasicMetadataSchema*, is an instance of itself
+(meaning that it can be used to describe itself).
+Hence, in DLite is **everything is an instance**.
+This has a practical implication that the API for instances can be applied
+to all metadata as well.
+Since the BasicMetadataSchema can describe itself, no more abstraction
+levels are needed, making the DLite metadata schema complete and well-defined.
+
+From Figure 2, one can also see that *[collections]* are simply
+instances of the *CollectionEntity*.
+
+In DLite all instances that describe other instances (of a lower abstraction
+level) are called *metadata*.
+
+Compared to ontologies, *data instances* correspond to OWL individuals
+while *entities* correspond to OWL classes.
+OWL, which is based on first order logic, do not have concepts corresponding
+to the higher abstraction levels of EntitySchema and BasicMetadataSchema.
+
+A (so far unexplored possibility) with such a datamodel hierarchy is
+that it can enable cross-platform semantic interoperability between
+independent systems that describe their datamodel hierarchies using a
+common BasicMetadataSchema.
+Of course, this requires a common agreement on the BasicMetadataSchema.
+
+
+### Entities
+An entity can be a single thing or object that represents something
+physical or nonphysical, concretely or abstract.
+The entity contains information about the data that constitutes the
+state of thing it describes.
```

Comment:
What does "State of thing" mean? should it be 'the state of the object is describes'?

Reply:
Ja, det er mye bedre